great_expectations/experimental/datasources/postgres_datasource.py

```diff
@@ -83,21 +89,36 @@ def param_defaults(self, data_asset: DataAsset) -> Dict[str, List]:
             data_asset.datasource.execution_engine, SqlAlchemyExecutionEngine
         )
         with data_asset.datasource.execution_engine.engine.connect() as conn:
-            # We opt to use a raw string instead of sqlalchemy.txt because we don't
-            # need any of the features for this query
-            # https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.text
-            col = self.column_name
-            q = f"select min({col}), max({col}) from {data_asset.table_name}"
-            min_dt, max_dt = list(conn.execute(q))[0]
-        year: List[int] = list(range(min_dt.year, max_dt.year + 1))
+            datetimes: DatetimeRange = (
+                _get_sqlite_datetime_range(conn, table_name=data_asset.table_name, col_name=self.column_name)
+            ) if conn.dialect.name == 'sqlite' else (
+                _get_sql_datetime_range(conn, table_name=data_asset.table_name, col_name=self.column_name)
+            )
+        year: List[int] = list(range(datetimes.min.year, datetimes.max.year + 1))
         month: List[int]
-        if min_dt.year == max_dt.year:
-            month = list(range(min_dt.month, max_dt.month + 1))
+        if datetimes.min.year == datetimes.max.year:
+            month = list(range(datetimes.min.month, datetimes.max.month + 1))
         else:
             month = list(range(1, 13))
         return {"year": year, "month": month}
 
 
+# In _get_sql*, we opt to use a raw string instead of sqlalchemy.text because we don't
+# need any of the features for these queries:
+# https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.text
+def _get_sql_datetime_range(conn: sqlalchemy.engine.base.Connection, table_name:str, col_name:str) -> DatetimeRange:
+    q = f'select min({col_name}), max({col_name}) from {table_name}'
+    min_max_dt = list(conn.execute(q))[0]
+    return DatetimeRange(min=min_max_dt[0], max=min_max_dt[1])
+
+
+def _get_sqlite_datetime_range(conn: sqlalchemy.engine.base.Connection, table_name: str,
+                                   col_name: str) -> DatetimeRange:
+    q = f"select STRFTIME('%Y%m%d', min({col_name})), STRFTIME('%Y%m%d', max({col_name})) from {table_name}"
+    min_max_dt = [datetime.strptime(dt, "%Y%m%d") for dt in list(conn.execute(q))[0]]
+    return DatetimeRange(min=min_max_dt[0], max=min_max_dt[1])
```

Comment:
I hate that we are already needing to special case sqlite here. It is possible you could unify these methods by using the [sqlalchemy.func.date](https://stackoverflow.com/questions/16228195/comparing-dates-with-sqlalchemy-sqlite) built-in, as also mentioned [here](https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#date-and-time-types). This would require using the ORM instead of the raw string and would negate your comment on line 106.

Reply:
I am going to break up this class into a SQL datasource and Postgres datasource as my next bit of work. I could also add a sqlite one so we don't branch here.

This is my concern about using the ORM. I haven't used sqlalchemy very much so if this isn't an issue let me know. I don't know the type of `col_name`. When I specify the ORM model can I use a sqlalchemy`date` type even if its a `datetime` or other type at the database level? I'm also assuming my model and skip columns that are present in the original table? I hope to sidestep these questions by making specific SQL datasources as described above but if you have experience with this let me know.